### DISCUSSION TITLE
In-dialog OPTIONS support

### DISCUSSION THREAD

**Discussion by bwalden** [resiprocate/resiprocate#189] [Ideas]
All OPTIONS requests are currently handled as out-of-dialog transactions. Unless a custom OutOfDialogHandler is supplied to the DialogUsageManager, any OPTIONS request is automatically met with a 200 response.

I have a use case where it would be beneficial to respond 481 to an in-dialog OPTIONS request, if that dialog does not exist.

**Comment by sgodin** [comment]
Looking at the code in ialogUsageManager::processRequest .... if the OPTIONs request has a To tag (ie in-dialog), then it will try to lookup the DialogSet, and it will respond with a 481 if that dialogset doesn't exist (line 2033).
